File: app/src/main/java/com/example/ls/maillistdemo/maillist/PinyinComparator.java
package com.example.ls.maillistdemo.maillist;

import java.util.Comparator;

/**
 * Created by LS on 2017/12/29.
 */

public class PinyinComparator implements Comparator<MedicineBean> {

    public int compare(MedicineBean o1, MedicineBean o2) {
        if (o1.getLetter().equals("@") || o2.getLetter().equals("#")) {
            return -1;
        } else if (o1.getLetter().equals("#")
                || o2.getLetter().equals("@")) {
            return 1;
        } else {
            return o1.getLetter().compareTo(o2.getLetter());
        }
    }
}

File: app/src/main/java/com/example/ls/maillistdemo/maillist/PinyinComparator.kt
package com.example.ls.maillistdemo.maillist

import java.util.*

/**
 * Created by LS on 2017/12/29.
 */

class PinyinComparator : Comparator<MedicineBean> {

    override fun compare(o1: MedicineBean, o2: MedicineBean): Int {
        return if (o1.letter == "@" || o2.letter == "#") {
            -1
        } else if (o1.letter == "#" || o2.letter == "@") {
            1
        } else {
            o1.letter!!.compareTo(o2.letter!!)
        }
    }
}
